src/Orchard.Cms.Web/Modules/Orchard.OpenId/Drivers/OpenIdSettingsDisplayDriver.cs

```diff
@@ -14,18 +17,33 @@ namespace Orchard.OpenId.Drivers
 {
     public class OpenIdSiteSettingsDisplayDriver : SiteSettingsSectionDisplayDriver<OpenIdSettings>
     {
+        private const string RestartPendingCacheKey = "OpenIdSiteSettings_RestartPending";
+        private const string SettingsGroupId = "open id";
+
         private readonly IOpenIdService _openIdServices;
         private readonly IHttpContextAccessor _httpContextAccessor;
+        private readonly INotifier _notifier;
+        private readonly IHtmlLocalizer<OpenIdSiteSettingsDisplayDriver> T;
+        private readonly IMemoryCache _memoryCache;
 
         public OpenIdSiteSettingsDisplayDriver(IOpenIdService openIdServices,
-                                                IHttpContextAccessor httpContextAccessor)
+                                                IHttpContextAccessor httpContextAccessor,
+                                                INotifier notifier,
+                                                IHtmlLocalizer<OpenIdSiteSettingsDisplayDriver> stringLocalizer,
+                                                IMemoryCache memoryCache)
         {
             _openIdServices = openIdServices;
+            _notifier = notifier;
             _httpContextAccessor = httpContextAccessor;
+            _memoryCache = memoryCache;
+            T = stringLocalizer;
         }
 
         public override IDisplayResult Edit(OpenIdSettings settings, BuildEditorContext context)
         {
+            if (context.GroupId == SettingsGroupId && _memoryCache.Get(RestartPendingCacheKey)!=null)
+                _notifier.Warning(T["This settings only will take effect after restarting the tenant."]);
```

Comment:
Not good English, so here is another suggestion: "The site needs to be restarted for the settings to take effect". 
I don't think the word tenant should be used here as it might confuse the user.